### PR TITLE
[6CORE-029] created hexagonal annotation markers.

### PR DESCRIPTION
**Pull request by YuraKhilko**
[6CORE-029] created hexagonal annotation markers.

### CODE REVIEW

File: 2-library/hexagonal-lib/pom.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.6core.lib</groupId>
+    <artifactId>hexagonal-lib</artifactId>
```
Comment (sokimaaa) [inline 2-library/hexagonal-lib/pom.xml]: bad naming of groupId + artifactId. 
Why do we need postfix `lib` if we already have it in groupdId? 
hexagonal-markers, hexagonal-annotations, hexagonal-specifications, ....
```suggestion
    <groupId>com.6core.lib</groupId>
    <artifactId>hexagonal-lib</artifactId>
```

File: 2-library/hexagonal-lib/pom.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    <version>0.0.1-SNAPSHOT</version>
+
+    <properties>
+        <java.version>17</java.version>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
```
Comment (sokimaaa) [inline 2-library/hexagonal-lib/pom.xml]: will code work without this properties?
```suggestion
```

File: 2-library/hexagonal-lib/src/main/java/com/_6core/lib/java/hexagonallib/v1/PortOut.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * @PortOut describes how our application can use other application/service.
+ * <p>
+ * Applicable for annotating an interface/class that describes outgoing interaction,
+ * e.g. interaction with DB layer (repository).
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target(ElementType.TYPE)
+public @interface PortOut {
```
Comment (sokimaaa) [inline 2-library/hexagonal-lib/src/main/java/com/_6core/lib/java/hexagonallib/v1/PortOut.java]: @DrivenAdapter and @DrivingAdapter missing.

**Review by sokimaaa — CHANGES_REQUESTED** [review]

File: 2-library/hexagonal-lib/src/main/java/com/_6core/lib/java/hexagonallib/v1/DomainService.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package com._6core.lib.java.hexagonallib.v1;
```
Comment (sokimaaa) [inline 2-library/hexagonal-lib/src/main/java/com/_6core/lib/java/hexagonallib/v1/DomainService.java]: We dont need versioning here. 
These annotations are going to be created once and forever.
Remember that package name is `groupId + artifactId`
```suggestion
package com._6core.lib.java.hexagonallib;
```

File: 2-library/hexagonal-lib/pom.xml (hunk 1 above)
Comment (YuraKhilko) [reply 2-library/hexagonal-lib/pom.xml]: Have changed to hexagonal-annotations.

File: 2-library/hexagonal-lib/pom.xml (hunk 2 above)
Comment (YuraKhilko) [reply 2-library/hexagonal-lib/pom.xml]: At local machine everything is ok, but in GitHub I've got an error connected with compilation. So I re added Java version.

**Pull request merged**